test/validation/api/ipsec/ipsec.c

```diff
@@ -534,7 +543,10 @@ static odp_event_t recv_pkt_async_inbound(odp_ipsec_op_status_t status)
 	 * queue specified during odp_ipsec_config()
 	 */
 	do {
-		event = odp_queue_deq(suite_context.default_queue);
+		if (odp_queue_type(suite_context.default_queue) == ODP_QUEUE_TYPE_PLAIN)
+			event = odp_queue_deq(suite_context.default_queue);
+		else
+			event = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
```

Comment:
This no longer checks that the event comes through the default queue, which is what the second patch added and the purpose of the second patch. Similarly, the reception through the destination queue does not check in scheduled mode that the queue indeed was the destination queue and not the default queue.

I propose using a common function for the event reception and passing the expected queue as a parameter. The function can check the queue type and use the appropriate dequeuing mechanism and in case of scheduled mode it can check that the input queue reported by odp_schedule() is the expected queue.

Reply:
Ack. Introduced new function recv_event(). Added check in it to test the queue from which the event was received for sched mode.